integration-libs/s4-service/checkout/core/facade/checkout-service-details.service.ts

```diff
@@ -0,0 +1,121 @@
+} from '@spartacus/core';
+import { Observable, combineLatest, take, map, switchMap, tap } from 'rxjs';
+import {
+  CheckoutServiceDetailsFacade,
+  CheckoutServiceDetailsSetEvent,
+} from '@spartacus/s4-service/root';
+import { CheckoutServiceDetailsConnector } from '../connector/checkout-service-details.connector';
+import { ServiceTime } from '@spartacus/s4-service/root';
+
+@Injectable()
+export class CheckoutServiceDetailsService
+  implements CheckoutServiceDetailsFacade
+{
+  protected activeCartFacade = inject(ActiveCartFacade);
+  protected userIdService = inject(UserIdService);
+  protected commandService = inject(CommandService);
+  protected serviceDetailsConnector = inject(CheckoutServiceDetailsConnector);
+  protected eventService = inject(EventService);
+  protected checkoutQueryFacade = inject(CheckoutQueryFacade);
+
+  protected setServiceScheduleSlotCommand: Command<string, unknown> =
+    this.commandService.create<string>(
+      (scheduledAt) =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) =>
+            this.serviceDetailsConnector
+              .setServiceScheduleSlot(userId, cartId, {
+                scheduledAt,
+              })
+              .pipe(
+                tap(() =>
+                  this.eventService.dispatch(
+                    {
+                      userId,
+                      cartId,
+                      scheduledAt,
+                    },
+                    CheckoutServiceDetailsSetEvent
+                  )
+                )
+              )
+          )
+        ),
+      {
+        strategy: CommandStrategy.CancelPrevious,
+      }
+    );
+
+  protected checkoutPreconditions(): Observable<[string, string]> {
+    return combineLatest([
+      this.userIdService.takeUserId(),
+      this.activeCartFacade.takeActiveCartId(),
+      this.activeCartFacade.isGuestCart(),
+    ]).pipe(
+      take(1),
+      map(([userId, cartId, isGuestCart]) => {
+        if (
+          !userId ||
+          !cartId ||
+          (userId === OCC_USER_ID_ANONYMOUS && !isGuestCart)
+        ) {
+          throw new Error('Checkout conditions not met');
+        }
+        return [userId, cartId];
+      })
+    );
+  }
+
+  setServiceScheduleSlot(scheduledAt: ServiceTime): Observable<unknown> {
+    return this.setServiceScheduleSlotCommand.execute(scheduledAt);
+  }
+
+  getSelectedServiceDetailsState(): Observable<QueryState<string | undefined>> {
+    return this.checkoutQueryFacade.getCheckoutDetailsState().pipe(
+      switchMap((state) => {
+        return this.getServiceProducts().pipe(
+          map((products) => {
+            if (products.length > 0) {
+              return { ...state, data: state.data?.servicedAt };
+            } else {
+              return { ...state, data: undefined };
+            }
+          })
+        );
+      })
+    );
+  }
+
+  getServiceProducts(): Observable<string[]> {
+    return this.activeCartFacade.getEntries().pipe(
```

Comment:
if any error is thrown anywhere in the methods, it will be handled at the end in component, and "An unknown error occurred. Please contact support." message will be shown. Similar to how it is done in all other checkout tabs.